docs/source/reference/integration.rst

```diff
@@ -148,3 +148,62 @@ XGBoost
    :nosignatures:
 
    optuna.integration.XGBoostPruningCallback
+
+Dependencies of each integration
+================================
```

Comment:
```suggestion
--------------------------------
```
It should be dashes, not equal signs. Otherwise, the title appears in an unintended place in the sidebar.
<img width="292" alt="image" src="https://github.com/optuna/optuna/assets/1255726/dcaa12b2-30e5-4200-b86a-bc0dfe899902">